apis/python/src/tiledbsoma/util.py

```diff
@@ -113,17 +113,24 @@ def uri_joinpath(base: str, path: str) -> str:
     return urllib.parse.urlunparse(parts)
 
 
-def slice_to_range(ids: slice) -> Optional[Tuple[int, int]]:
+def slice_to_range(
+    ids: slice, nonempty_domain: Tuple[int, int]
+) -> Optional[Tuple[int, int]]:
     """
     For the interface between ``DataFrame::read`` et al. (Python) and ``SOMAReader`` (C++).
     """
+    if ids.step is not None and ids.step != 1:
+        raise ValueError("slice step must be 1 or None")
     if ids.start is None and ids.stop is None:
         return None
-    if ids.start is None or ids.stop is None:
-        # TODO: https://github.com/single-cell-data/TileDB-SOMA/issues/457
-        raise ValueError("slice start and stop must be both specified, or neither")
-    if ids.start > ids.stop:
+    start = ids.start
+    stop = ids.stop
+
+    if start is None:
```

Comment:
Just a thought ... maybe add a to-do comment that with more work on the C++ reader we wouldn't need to do this in Python ...